### PR TITLE
Make the `api_key` constructor parameter Optional and read from env

### PR DESCRIPTION
**Pull request by sharvil**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ async def main():
 asyncio.run(main())
 ```
 
+While you can provide an `api_key` argument, we recommend using python-dotenv to add LMNT_API_KEY="My API Key" to your .env file so that your API key is not stored in source control.
```
Comment (shaper) [inline README.md:43]: nit: back-ticks around `python-dotenv`, `LMNT_API_KEY="My Api Key"`, and `.env`

File: src/lmnt/api.py [hunk 2]
```diff
@@ -121,11 +123,13 @@ async def finish(self):
...
 
 class Speech:
-  def __init__(self, api_key: str, **kwargs):
+  def __init__(self, api_key: Optional[str] = None, **kwargs):
     self._session = None
-    self._api_key = api_key
+    self._api_key = api_key or os.environ.get('LMNT_API_KEY')
     self._base_url = kwargs.get('base_url', _BASE_URL)
     self._connector = kwargs.get('connector', None)
+    if self._api_key is None:
```
Comment (shaper) [inline src/lmnt/api.py:131]: I think an empty string will pass this -- perhaps `if not self._api_key:` instead?

**Review by shaper — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (sharvil) [reply README.md]: Done.

File: src/lmnt/api.py (hunk 2 above)
Comment (sharvil) [reply src/lmnt/api.py]: Done.

**Pull request merged**